### ISSUE TITLE
Unable to click certain link in Capybara 2.5.0 with Selenium Driver in Firefox

### ISSUE THREAD

**Issue by bytenel** [jnicklas/capybara#1616]
Not sure what's up, I think it's a bug in Capybara as I can't otherwise determine what I'm doing wrong here.

I recently upgraded Rails from 4.1.0 to 4.2.5 and then this particular test started failing:

```
 it 'user can edit business info' do
    user = create(:editor_user, :with_business)
    login user

    visit business_info_path(user.business)
    click_link 'Edit'

    fill_in 'Account Contact', with: 'name'
    click_button 'Update'
    expect(page).to have_content 'Your business information has been updated.'
    expect(page).to have_content 'name'
    expect(user.business.reload.account_contact).to eq 'name'
  end
```

The browser window loads appropriately it just never finds the Edit link that is present on the page. All the actual functionality appears to work appropriately. If I just have a visit to that path everything works afterwards as well. For some reason it just doesn't like that link.

The markup is below for that link.

```
<div class="row">
  <div class="columns small-12 medium-6">
    <div class="key-value">
      <div class="key">Actions</div>
      <div class="value">
        <%= link_to 'Edit', edit_business_info_path(@business), class: 'btn btn-mini btn-warning', id: 'edit' %>
      </div>
    </div>
  </div>
</div>
```

**Comment by twalpole** [comment]
What is the actual html when the page is open (rather than the erb), and what exactly is the error you're getting

**Comment by bytenel** [comment]
Error I am getting is:

```
  Capybara::ElementNotFound:
       Unable to find field "Account Contact"
```
Which makes sense because the click_link to the edit should load the edit page which has that form in it.

The html is below.

```
<div class="row">
    ::before
    <div class="columns small-12 medium-6">
        <div class="key-value">
            <div class="key"></div>
            <div class="value">
                <a id="edit" class="btn btn-mini btn-warning" href="/businesses/renner-mckenzie/info/edit"></a>
            </div>
        </div>
    </div>
    ::after
</div>
```

The strange part is when I have a binding.pry added to the page before the click_link call (so I could see what renders) and then continue execution the test seems to work. Is it an async problem related to the visit call?

**Comment by twalpole** [comment]
Does the page a user is directed to after logging in have a link with 'Edit' in it??  It is possible that you are actually clicking the link on that page before the visit call has actually changed to the business_info_path(user.business) page.  You can check that by doing 

    expect(page).to have_current_path(business_info_path(user.business)) 

after the visit, or an expectation for text/elements that would only be on that page.  You could also change to 

   click_link('Edit', href: edit_business_info_path(user.business))

to make sure it's clicking the correct Edit link

**Comment by bytenel** [comment]
Awesome, thank you! It does actually have a link on the page with that text in it...that is very strange now that I double check on that rendered HTML that it is not present there...that would explain at least what I thought was voodoo. Now the crazy thing is why is it rendering when I see the actual page when the test harness runs but not in the HTML. 

I'll try those steps and report back what I find. In the interim I tried fiddling with my updated gems to make sure nothing was being caused by some other dependency in the project being updated, seems like I got other failures so that could have been it as well. I appreciate the support!
